ast/src/analyzed/mod.rs

```diff
@@ -826,6 +871,9 @@ pub struct PolynomialReference {
     /// Optional because it is filled in in a second stage of analysis.
     /// TODO make this non-optional
     pub poly_id: Option<PolyID>,
+    /// The type arguments if the symbol is generic.
+    /// Might be empty before type checking is completed.
```

Comment:
This sounds a bit sketchy. Shouldn't it use a different type then? Or be a different type completely, pre and post type checking?

Reply:
We had that before and it was a big mess especially with the assembly that is still not properly integrated. Would prefer to keep it an option for now.